doc-content/enterprise-only/optimizer/planner-quarkus-database-proc.adoc

```diff
@@ -0,0 +1,156 @@
+
+    @Inject
+    SolverManager<TimeTable, Long> solverManager;
+    @Inject
+    ScoreManager<TimeTable, HardSoftScore> scoreManager;
+
+    // To try, open http://localhost:8080/timeTable
+    @GET
+    public TimeTable getTimeTable() {
+        // Get the solver status before loading the solution
+        // to avoid the race condition that the solver terminates between them
+        SolverStatus solverStatus = getSolverStatus();
+        TimeTable solution = findById(SINGLETON_TIME_TABLE_ID);
+        scoreManager.updateScore(solution); // Sets the score
+        solution.setSolverStatus(solverStatus);
+        return solution;
+    }
+
+    @POST
+    @Path("/solve")
+    public void solve() {
+        solverManager.solveAndListen(SINGLETON_TIME_TABLE_ID,
+                this::findById,
+                this::save);
+    }
+
+    public SolverStatus getSolverStatus() {
+        return solverManager.getSolverStatus(SINGLETON_TIME_TABLE_ID);
+    }
+
+    @POST
+    @Path("/stopSolving")
+    public void stopSolving() {
+        solverManager.terminateEarly(SINGLETON_TIME_TABLE_ID);
+    }
+
+    @Transactional
+    protected TimeTable findById(Long id) {
+        if (!SINGLETON_TIME_TABLE_ID.equals(id)) {
+            throw new IllegalStateException("There is no timeTable with id (" + id + ").");
+        }
+        // Occurs in a single transaction, so each initialized lesson references the same timeslot/room instance
+        // that is contained by the timeTable's timeslotList/roomList.
+        return new TimeTable(
+                Timeslot.listAll(Sort.by("dayOfWeek").and("startTime").and("endTime").and("id")),
+                Room.listAll(Sort.by("name").and("id")),
+                Lesson.listAll(Sort.by("subject").and("teacher").and("studentGroup").and("id")));
+    }
+
+    @Transactional
+    protected void save(TimeTable timeTable) {
+        for (Lesson lesson : timeTable.getLessonList()) {
+            // TODO this is awfully naive: optimistic locking causes issues if called by the SolverManager
+            Lesson attachedLesson = Lesson.findById(lesson.getId());
+            attachedLesson.setTimeslot(lesson.getTimeslot());
+            attachedLesson.setRoom(lesson.getRoom());
+        }
+    }
+
+}
+----
++
+This example includes a `TimeTable` instance. However, you can enable multi-tenancy and handle `TimeTable` instances for multiple schools in parallel.
++
+The `getTimeTable()` method returns the latest timetable from the database. It uses the `ScoreManager` method, which is automatically injected, to calculate the score of that timetable and make it available to the UI.
++
+The `solve()` method starts a job to solve the current timetable and stores the time slot and room assignments in the database. It uses the `SolverManager.solveAndListen()` method to listen to intermediate best solutions and update the database accordingly. This enables the UI to show progress while the backend is still solving.
```

Comment:
```suggestion
The `solve()` method starts a job to solve the current timetable and stores the time slot and room assignments in the database. It uses the `SolverManager.solveAndListen()` method to listen to intermediate best solutions and update the database accordingly. The UI can use this to show progress while the backend is still solving.
```

Remove [anthropomorphism](https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism). Also, do we know the specific part of the UI that will show this? See [user interface](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#u).